### DISCUSSION TITLE
Logbook message is added as json formatted string instead of json object

### DISCUSSION THREAD

**Discussion by drobev** [logfellow/logstash-logback-encoder#817] [General]
Hi, 
I'm on a spring boot 2.6.6 stack with logback for logging. I'm trying to get the logging right for ES and Kibana. Unfortunately I'm facing the issue that the logbook logs aren't wwritten as json object but as json formatted string. This way they can't be used in kibana queries.

config:
`<configuration>

    
    <shutdownHook class="ch.qos.logback.core.hook.DelayingShutdownHook"/>

    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
            <jsonFactoryDecorator class="net.logstash.logback.decorate.CharacterEscapesJsonFactoryDecorator">
                <includeStandardAsciiEscapesForJSON>false</includeStandardAsciiEscapesForJSON>
            </jsonFactoryDecorator>
            <timeZone>UTC</timeZone>
            <timestampPattern>yyyy-MM-dd' 'HH:mm:ss.SSSzzz</timestampPattern>
        </encoder>
    </appender>

    <root level="DEBUG">
        <appender-ref ref="CONSOLE"/>
    </root>

</configuration>`

This creates logs like
`{"@timestamp":"2022-07-04 14:27:51.174UTC","@version":"1","message":"Received multipartFile...}`
which are fine.
And logs, where the message is already a json, which aren't fine:
`{"@timestamp":"2022-07-04 14:27:51.191UTC","@version":"1","message":"{"origin":"local","type":"response", ...}`
As you can see the message here starts with " and is not recognized as json. 
Any hints, how can I configure the appender (or use some other appender).
Or which spring bean is responsible for this. So I can replace it with some custom logic?

Best, 
Dimitar

**Comment by brenuart** [comment]
Your issue looks similar to https://github.com/logfellow/logstash-logback-encoder/issues/264.
Does it help ?

**Reply by drobev** [reply]
Hi brenuart, 
10x for the reply.
Unfortunately not. I tried what is sugested there but without any success. I got an 'missing field' error, which i wasn't able to solve.
Initially I had the probelm with escaped " as well, but I solved that one via
`        <jsonFactoryDecorator class="net.logstash.logback.decorate.CharacterEscapesJsonFactoryDecorator">
            <includeStandardAsciiEscapesForJSON>false</includeStandardAsciiEscapesForJSON>
        </jsonFactoryDecorator>
`
The problem that JSON messages are inserted as a string still remains.

Cheers!
Dimitar

**Reply by brenuart** [reply]
It looks like your application code is pre-converting an object to a JSON string, and then passing that string to the logger as the log message. This is not an expected/supported use case as @philsttr mentioned in the linked issue.

Phil mentioned several alternatives... which one did you try and is still causing you trouble? Could you please post a sample configuration together with a typical logging statement and the associated payload?
